src/crop-manager.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+interface VatLike {
+    function live() external view returns (uint256);
+    function urns(bytes32, address) external view returns (uint256, uint256);
+    function dai(address) external view returns (uint256);
+    function fork(bytes32, address, address, int256, int256) external;
+    function frob(bytes32, address, address, address, int256, int256) external;
+    function flux(bytes32, address, address, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface CropLike {
+    function gem() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function join(address, address, uint256) external;
+    function exit(address, address, uint256) external;
+    function tack(address, address, uint256) external;
+    function flee(address) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external;
+    function transferFrom(address, address, uint256) external;
+}
+
+contract UrnProxy {
+    address immutable public vat;
+    address immutable public usr;
+    address public owner;
+
+    constructor(address vat_, address usr_) public {
+        owner = msg.sender;
+        vat = vat_;
+        usr = usr_;
+        VatLike(vat_).hope(msg.sender);
+    }
+
+    function migrate(address newOwner) external {
```

Comment:
To be useful, I think there needs to be a function on the manager that calls this. However, now that we have `quit` (which should solve the ES issue), I'm am not certain there's a separate use case for this.

Reply:
The idea was to leave this open for future implementation changes if we needed to delegate to another address, but since the implementation is upgradable and it's hoped, I'm not sure this is necessary either.

Reply:
Conclusion: will remove `migrate` and `owner`.

Reply:
It was done.